### PR TITLE
Developer Guide Extension

### PR DESCRIPTION
**Pull request by SiHaoShen**
- Updated all the open source software (Grafana, InfluxDB, NodeRed)
- Changed NodeRed flows due to software update
- Add additional security aspect in docker-compose.yml: Linux capabilities, read-only file system, non-root user, container initialization

### CODE REVIEW

File: src/solution/my_edge_app/docker-compose.yml [hunk 1]
```diff
@@ -33,27 +33,33 @@ services:
       - grafana                         # Wait for start of container 'grafana'
 
 ##### INFLUXDB ######
+  init-db:
+    image: init-db:v0.0.1               # Docker Init Container to Setup the Main InfluxDB v2.2 Container
```
Comment (EndyCZ) [inline src/solution/my_edge_app/docker-compose.yml:37]: Please add build statement with a path https://docs.docker.com/compose/compose-file/compose-file-v2/#build 
the goal is that user of the guide can just directly run docker-compose up -d commad after runnign the  [mosquite docker copose file](src/solution/mqtt_broker_mosquitto/docker-compose.yml)

File: src/solution/my_edge_app/docker-compose_Edge.yml [hunk 2]
```diff
@@ -26,23 +26,32 @@ services:
...
     volumes:                            # mount volume from host
       - db-backup:/var/lib/influxdb     # mount named volume 'db-backup' to host's path to /var/lib/influxdb
+    healthcheck:
+      test: "exit 0"
+    networks:                           # define networks connected to container 'influxdb' 
+      proxy-redirect:                   # Name of the network
+  
+  init-db:
+    image: init-db:v0.0.1               # Docker Init Container to Setup the Main InfluxDB v2.2 Container
+    mem_limit: 100m
```
Comment (EndyCZ) [inline src/solution/my_edge_app/docker-compose_Edge.yml:49]: Please add build statement with a path https://docs.docker.com/compose/compose-file/compose-file-v2/#build 
the goal is that user of the guide can just directly run docker-compose up -d commad after runnign the  [mosquite docker copose file](src/solution/mqtt_broker_mosquitto/docker-compose.yml)

File: src/solution/my_edge_app/influxdb/init_influxdb.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -x
+set -e
+
+
```
Comment (EndyCZ) [inline src/solution/my_edge_app/influxdb/init_influxdb.sh:4]: Add comment that in this file we would load secrets from various Secrement management sources and use them to confugre the influx db database.

**Review by EndyCZ — CHANGES_REQUESTED** [review]

**Review by EndyCZ — CHANGES_REQUESTED** [review]
I did not found any folder src/solution/tracee which you describes.

**Review by EndyCZ — CHANGES_REQUESTED** [review]
Please also remove all the $http_proxy and $https_proxy. I think they are not needed anymore from the docker-compose files

**Comment by SiHaoShen** [comment]
# Feedback from Review of 08.01
I removed all the $http_proxy and $https_proxys variable. Also I added build statement in **docker-compose.yml**, but not in **docker-compose_Edge.yml**. Since build statement is not supported in edge app publisher

**Comment by EndyCZ** [comment]
Cannot import into IEAP image with enviroment variable. https://github.com/SiHaoShen/Developer-Guide-Hands-on-App/blob/main/src/solution/HandsOn_1/my_edge_app/docker-compose_Edge.yml#L29

**Review by EndyCZ — APPROVED** [review]
Merging now into dev branch to have it availalbe in organization repository. Could not merge because of Software clearing and some small details to adjust.

**Pull request merged**